src/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,29 @@
+import classNames from 'classnames';
+import { useContext } from 'react';
+import { PostDetails } from '../PostData/PostDetails';
+import { StateContext } from '../../reducer/store';
+
+export const Sidebar = () => {
+  const { selectedPost } = useContext(StateContext);
```

Comment:
Also, remember that useContext hook will only return the current context value, not the whole context object. Ensure that you have provided a context value that contains the selectedPost property.